src/gc/gc.cpp

```diff
@@ -26685,7 +26683,7 @@ void gc_heap::background_grow_c_mark_list()
         assert (new_c_mark_list);
         memcpy (new_c_mark_list, c_mark_list, c_mark_list_length*sizeof(uint8_t*));
         c_mark_list_length = c_mark_list_length*2;
-        delete c_mark_list;
+        delete [] c_mark_list;
         c_mark_list = new_c_mark_list;
```

Comment:
Wow, didn't realize that would affect anything (seems like an unnecessary complexity for the language :P). Thanks! I see @janvorli changed a bunch of things from new (nothrow) (some_type[n]) to new (nothrow) some_type[n] a while ago that included the resizing of c_mark_list but not for the initial creation. It would be good to be consistent here - so we should never use new (nothrow) (some_type[n]) and always new (nothrow) some_type[n]. We can do this all in the issue that changes delete to delete[].